### PR TITLE
Add the Social Premium plan and enhanced publishing feature

### PR DESCRIPTION
**Pull request by gmjuhasz**
#### Proposed Changes

This change introduces the new enhanced sharing feature for social and adds the premium yearly and monthly plans to Calypso. It also fixes the naming convention for the 1000 shares feature.

#### Testing Instructions

Needs 1203313664893883-as-1203207028774795 merged before testing

* Head to http://calypso.localhost:3000/checkout/jetpack/jetpack_social_advanced_yearly to add Jetpack Social Premium Yearly to your shopping cart
* Verify that two variants appear (yearly and monthly)
* Verify that you can switch between the variants
 * Verify that you can complete the purchase using credits or the store sandbox.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: packages/calypso-products/src/products-list.ts [hunk 1]
```diff
@@ -356,10 +359,32 @@ export const JETPACK_SITE_PRODUCTS_WITH_FEATURES: Record<
 		term: TERM_MONTHLY,
 		bill_period: PLAN_MONTHLY_PERIOD,
 		categories: [],
-		getFeatures: () => [ SOCIAL_SHARES_1000 ],
+		getFeatures: () => [ FEATURE_SOCIAL_SHARES_1000 ],
 		getProductId: () => 2504,
 		getStoreSlug: () => PRODUCT_JETPACK_SOCIAL_BASIC_MONTHLY,
 	},
+	[ PRODUCT_JETPACK_SOCIAL_ADVANCED ]: {
+		product_name: translate( 'Social Premium' ),
```
Comment (danielpost) [inline packages/calypso-products/src/products-list.ts]: Should this be Social Advanced instead?

File: packages/calypso-products/src/products-list.ts [hunk 2]
```diff
@@ -356,10 +359,32 @@ export const JETPACK_SITE_PRODUCTS_WITH_FEATURES: Record<
...
+		type: PRODUCT_JETPACK_SOCIAL_ADVANCED,
+		term: TERM_ANNUALLY,
+		bill_period: PLAN_ANNUAL_PERIOD,
+		categories: [],
+		getFeatures: () => [ FEATURE_SOCIAL_SHARES_1000, FEATURE_SOCIAL_ENHANCED_PUBLISHING ],
+		getProductId: () => 2600,
+		getStoreSlug: () => PRODUCT_JETPACK_SOCIAL_ADVANCED,
+	},
+	[ PRODUCT_JETPACK_SOCIAL_ADVANCED_MONTHLY ]: {
+		product_name: translate( 'Social Premium' ),
```
Comment (danielpost) [inline packages/calypso-products/src/products-list.ts]: Same here.

**Review by danielpost — COMMENTED** [review]
Nice! 👍🏼  I'll give it a proper test as well, but here's one small thing I noticed while looking at the code.

**Review by spsiddarthan — APPROVED** [review]
This tests well, @gmjuhasz ! You might want to rebase it before merging.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7629430

Hi @gmjuhasz, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Social Advanced`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7629430).

**Pull request merged**